src/js/components/TextInput/TextInput.js

```diff
@@ -114,6 +114,21 @@ const TextInput = forwardRef(
     const [focus, setFocus] = useState();
     const [showDrop, setShowDrop] = useState();
 
+    const handleSuggestionSelect = useRef();
+    const handleTextSelect = useRef();
+
+    // assigns select handlers
+    useEffect(() => {
+      if (onSelect && !onSuggestionSelect) {
+        handleSuggestionSelect.current = onSelect;
+      } else if (!onSelect && onSuggestionSelect) {
+        handleSuggestionSelect.current = onSuggestionSelect;
+      } else {
+        handleSuggestionSelect.current = onSuggestionSelect;
```

Comment:
```suggestion
        // in case both onSelect & onSuggestionSelect are defined
        handleSuggestionSelect.current = onSuggestionSelect;
```
Adding descriptive comment.